helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -244,6 +244,10 @@ public Response updateCluster(@PathParam("clusterId") String clusterId,
     ClusterSetup clusterSetup = getClusterSetup();
     HelixAdmin helixAdmin = getHelixAdmin();
 
+    if (!doesClusterExist(clusterId)) {
```

Comment:
Will this cause an issue if the caller is attempting to use the "activate" command? What is the response if the cluster isn't activated?

Reply:
Good catch. Mentioned line will be removed.